cocos/core/pipeline/pipeline-ubo.ts

```diff
@@ -211,72 +211,74 @@ export class PipelineUBO {
         const linear = 0.0;
         const packing = supportsR32FloatTexture(device) ? 0.0 : 1.0;
 
-        if (shadowInfo.enabled && mainLight && mainLight.shadowEnabled) {
+        if (mainLight && shadowInfo.enabled) {
             if (shadowInfo.type === ShadowType.ShadowMap) {
```

Comment:
It is used for the protection judgment of shadow map. Planar shadow info is masked incorrectly